test/test_layers/test_learnable_squeezer.jl

```diff
@@ -0,0 +1,99 @@
+using InvertibleNetworks, LinearAlgebra, Test, Flux, Random
+device = InvertibleNetworks.CUDA.functional() ? gpu : cpu
+Random.seed!(11)
+
+
+# Dimensions
+n = (2*17, 3*11, 4*7)
+nc = 4
+batchsize = 3
+k = (2, 3, 4)
+
+for N = 1:3
+
+    # Test invertibility
+    C = LearnableSqueezer(k[1:N]...) |> device
+    X = randn(Float32, n[1:N]..., nc, batchsize) |> device
+    Y = randn(Float32, div.(n, k)[1:N]..., prod(k[1:N])*nc, batchsize) |> device
+    @test X ≈ C.inverse(C.forward(X)) rtol=1f-6
+    @test Y ≈ C.forward(C.inverse(Y)) rtol=1f-6
+
+
+    # Test backward/inverse coherence
+    ΔY = randn(Float32, div.(n, k)[1:N]..., prod(k[1:N])*nc, batchsize) |> device
+    Y  = randn(Float32, div.(n, k)[1:N]..., prod(k[1:N])*nc, batchsize) |> device
+    X_ = C.inverse(Y)
+    _, X = C.backward(ΔY, Y)
+    @test X ≈ X_ rtol=1f-6
+
+
+    # Test backward_inv/forward coherence
+    ΔX = randn(Float32, n[1:N]..., nc, batchsize) |> device
+    X  = randn(Float32, n[1:N]..., nc, batchsize) |> device
+    Y_ = C.forward(X)
+    _, Y = C.backward_inv(ΔX, X)
+    @test Y ≈ Y_ rtol=1f-6
+
+
+    # Gradient test (input)
+    ΔY = randn(Float32, div.(n, k)[1:N]..., prod(k[1:N])*nc, batchsize) |> device
+    ΔX = randn(Float32, n[1:N]..., nc, batchsize) |> device
+    X  = randn(Float32, n[1:N]..., nc, batchsize) |> device
+    Y = C.forward(X)
+    ΔX_, _ = C.backward(ΔY, Y)
+    @test dot(ΔX, ΔX_) ≈ dot(C.forward(ΔX), ΔY) rtol=1f-4
+
+
+    # Gradient test (input, inv)
+    ΔY = randn(Float32, div.(n, k)[1:N]..., prod(k[1:N])*nc, batchsize) |> device
+    ΔX = randn(Float32, n[1:N]..., nc, batchsize) |> device
+    Y = randn(Float32, div.(n, k)[1:N]..., prod(k[1:N])*nc, batchsize) |> device
+    X = C.inverse(Y)
+    ΔY_, _ = C.backward_inv(ΔX, X)
+    @test dot(ΔY, ΔY_) ≈ dot(C.inverse(ΔY), ΔX) rtol=1f-4
+
+
+    # Gradient test (parameters)
+    T = Float64
+    C = LearnableSqueezer(k[1:N]...) |> device; C.stencil_pars.data = InvertibleNetworks.CUDA.cu(C.stencil_pars.data)
```

Comment:
Those explicit calls to CUDA will break ci. Need to use device here even if not as optimal

Reply:
One issue that I have with |> device is the automatic conversion to Float32; here I was using Float64 to avoid numerical cancellation to spoil the gradient test. Anyway, I will rewrite the test so that we are more consistent with the rest of the tests.

Related to this but perhaps for later, I think it might be a good idea to make every InvertibleNetwork subtype parametric. It might make these conversions and include half-precision nets more easily. Unless there are more important reasons not to do that.

Reply:
>Related to this but perhaps for later, I think it might be a good idea to make every InvertibleNetwork subtype parametric. 

Yes completely agree!